Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM ghcr.io/plc-lang/rust-llvm:latest
+
+# Allow invoking `plc` from anywhere
+ENV PLCLOC="/opt/rusty"
+ENV STDLIBLOC="/opt/rusty/stdlib"
+ENV PATH="${PLCLOC}:${PATH}"
+
+# Give all users access to cargo and rust home
+RUN chmod -R a+rw $CARGO_HOME \
+    && chmod -R a+rw $RUSTUP_HOME
+
+# Required if we want to use `lld` as the default linker for RuSTy
+ENV LLVM_VER=14
+RUN ln -sf /usr/bin/ld.lld-$LLVM_VER /usr/bin/ld.lld
+
+# Install the local RuSTy version
+WORKDIR /rusty
+COPY . .
+RUN ./scripts/build.sh --build --release && \
```

Comment:
You should use `--package` option here, the so that rust produces is missing some symbols, I was never able to figure out how to link correctly using the build.rs (`--as-needed` did not behave correctly). With the `--package` option the lib will be automatically compiled and placed in the project in the folder called output. It will be split into a include/ and a lib/ folder
The lib will contain a version for each target you compile for.
You can add `--target=x86_64-unknown-linux-gnu` and `--target=aarch64-unknown-linux-gnu` to compile the std lib for both targets at the same time. So the compiler will be x86_64 only but the lib can be used to target arm64 cpus

Reply:
I updated to the branch to use --package